Modules/_threadmodule.c

```diff
@@ -950,6 +950,7 @@ lock_new_impl(PyTypeObject *type)
     if (self == NULL) {
         return NULL;
     }
+    _PyObject_SetDeferredRefcount((PyObject *)self);
```

Comment:
Can you move the call after initializing `self->lock`? Same remark for `rlock_new_impl()` below.